pennylane/pytrees.py

```diff
@@ -54,5 +105,93 @@ def register_pytree(pytree_type: type, flatten_fn: FlattenFn, unflatten_fn: Unfl
 
     """
 
+    _register_pytree_with_pennylane(pytree_type, flatten_fn, unflatten_fn)
+
     if has_jax:
         _register_pytree_with_jax(pytree_type, flatten_fn, unflatten_fn)
+
+
+class Structure(namedtuple("Structure", ["type", "metadata", "children"])):
```

Comment:
`Structure` is a very vague name for a class. Imo something like `TreeDef` or `TreeStructure` would be more appropriate.

Reply:
Jax uses `PyTreeDef`, so we could repurpose that name.  But I think I prefer `TreeStructure`. I think the word "structure" better communicates how this is the layout of the pytree.

Reply:
I like `TreeStructure` as well :)